src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -471,39 +468,75 @@ def _post_process(
     upload_env_key: str,
     workspace_id: int,
     load_id: str,
-    obj_tuple: _WSObjTuple,
     upload_dir: str,
     output_dir: str,
-    upa: str,
+    assembly_tuple: _WSObjTuple,
+    assembly_upa: str,
+    genome_tuple: _WSObjTuple = None,
+    genome_upa: str = None,
+    assembly_obj_info: list[Any] = None,
+    genome_obj_info: list[Any] = None,
 ) -> None:
     """
+    Update the uploaded.yaml file in the genome directory with the object name and upa info.
+
+    If genome_tuple and genome_upa are provided, the function will also:
     Create a standard entry in sourcedata/workspace for each object.
     Hardlink to the original object file in sourcedata to avoid duplicating the file.
-    Update the uploaded.yaml file in the genome directory with the object name and upa info.
     Creates a softlink from new_dir in collectionssource to the contents of target_dir in sourcedata.
     """
-    # Create a standard entry in sourcedata/workspace
-    # hardlink to the original object file in sourcedata
-    src_file = _get_source_file(
-        obj_tuple.host_file_dir,
-        obj_tuple.obj_name,
-    )
-    target_dir = os.path.join(output_dir, upa)
-    os.makedirs(target_dir, exist_ok=True)
-    dest_file = os.path.join(target_dir, f"{upa}.fna.gz")
-    loader_helper.create_hardlink_between_files(dest_file, src_file)
-
     # Update the uploaded.yaml file
     _update_upload_status_yaml_file(
         upload_env_key,
         workspace_id,
         load_id,
-        upa,
-        obj_tuple,
+        assembly_upa,
+        assembly_tuple,
+        genome_upa=genome_upa,
+        genome_tuple=genome_tuple,
     )
 
+    if genome_tuple and genome_upa:
+
+        if not assembly_obj_info or not genome_obj_info:
+            raise ValueError(
+                "Both assembly_obj_info and genome_obj_info must be provided"
+            )
+
+        _process_genome_objects(
+            upload_dir, output_dir, assembly_tuple, assembly_upa, assembly_obj_info, genome_obj_info)
+
+
+def _process_genome_objects(
+        upload_dir: str,
+        output_dir: str,
+        assembly_tuple: _WSObjTuple,
+        assembly_upa: str,
+        assembly_obj_info: list[Any],
+        genome_obj_info: list[Any],
+) -> None:
+    """
+    Post process on successful genome uploads.
+    """
+    # Create a standard entry in sourcedata/workspace
+    # hardlink to the original object file in sourcedata
+    src_file = Path(_get_source_file(assembly_tuple.host_file_dir, assembly_tuple.obj_name))
+    target_dir = os.path.join(output_dir, assembly_upa)
+    os.makedirs(target_dir, exist_ok=True)
+
+    suffixes = src_file.suffixes
+    # TODO - handle extension other than .gz
+    dest_suffix = suffixes[-1] if suffixes[-1] != ".gz" else "".join(suffixes[-2:])
+    dest_file = os.path.join(target_dir, f"{assembly_upa}{dest_suffix}")
```

Comment:
We're assuming no archive files here right? So no *.tar.gz?

Reply:
no. only `.gz` file for now.

Reply:
:+1: